packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +356,267 @@ contract AssetCreate is
+    /// @param mintData The data for the lazy mint
+    function lazyCreateMultipleAssets(bytes memory signature, LazyMintBatchData calldata mintData)
+        external
+        whenNotPaused
+    {
+        require(
+            authValidator.verify(
+                signature,
+                _hashLazyBatchMint(
+                    mintData.creators,
+                    signatureNonces[_msgSender()]++,
+                    mintData.tiers,
+                    mintData.amounts,
+                    mintData.unitPrices,
+                    mintData.paymentTokens,
+                    mintData.metadataHashes,
+                    mintData.maxSupplies
+                )
+            ),
+            "AssetCreate: Invalid signature"
+        );
+
+        require(mintData.creators.length == mintData.tiers.length, "AssetCreate: 1-Array lengths");
+        require(mintData.amounts.length == mintData.metadataHashes.length, "AssetCreate: 2-Array lengths");
+        require(mintData.metadataHashes.length == mintData.tiers.length, "AssetCreate: 3-Array lengths");
+        require(mintData.tiers.length == mintData.unitPrices.length, "AssetCreate: 4-Array lengths");
+        require(mintData.unitPrices.length == mintData.paymentTokens.length, "AssetCreate: 5-Array lengths");
+        require(mintData.paymentTokens.length == mintData.maxSupplies.length, "AssetCreate: 6-Array lengths");
+
+        uint256[] memory tokenIds = new uint256[](mintData.tiers.length);
+        uint256[] memory tiersToBurn = new uint256[](mintData.tiers.length);
+        for (uint256 i = 0; i < mintData.tiers.length; i++) {
+            uint16 revealed = mintData.tiers[i] == 1 ? 1 : 0;
+            tokenIds[i] = assetContract.getTokenIdByMetadataHash(mintData.metadataHashes[i]);
+            if (tokenIds[i] == 0) {
+                tokenIds[i] = TokenIdUtils.generateTokenId(
+                    mintData.creators[i],
+                    mintData.tiers[i],
+                    ++creatorNonces[mintData.creators[i]],
+                    revealed,
+                    true
+                );
+                require(mintData.amounts[i] <= mintData.maxSupplies[i], "AssetCreate: Max supply exceeded");
+                availableToMint[tokenIds[i]] = mintData.maxSupplies[i] - mintData.amounts[i];
+            } else {
+                require(availableToMint[tokenIds[i]] >= mintData.amounts[i], "AssetCreate: Max supply reached");
+                availableToMint[tokenIds[i]] -= mintData.amounts[i];
+            }
+            distributePayment(
+                mintData.unitPrices[i],
+                mintData.amounts[i],
+                mintData.paymentTokens[i],
+                mintData.creators[i]
+            );
+        }
+
+        catalystContract.burnBatchFrom(_msgSender(), tiersToBurn, mintData.amounts);
+        assetContract.mintBatch(_msgSender(), tokenIds, mintData.amounts, mintData.metadataHashes);
```

Comment:
Do we need to consider setting a max length for the batch mint array?

Reply:
I think it will be enough to set it on the frontend + backend